src/Filter/FilterInterface.php

```diff
@@ -90,14 +90,14 @@ public function getParentAssociationMappings(): array;
     /**
      * @throws \RuntimeException if the `field_mapping` option is not set
      *
-     * @return array<string, string> field mapping
+     * @return array<string, mixed> field mapping
```

Comment:
Can we add an array shape signature here?

Reply:
I would say no because it's persistence bundle specific.

Reply:
Even if it is persistence bundle specific, this interface defines the contract. What options are normally returned here?

Reply:
I would recommend to discuss this in an issue like https://github.com/sonata-project/SonataAdminBundle/issues/6274 and not block this PR which is just a phpdoc fix.

The option returns can be anything, based on what the persistence bundle use. DoctrineORM and DoctrineMongo doesn't use the same key or values. For instance targetDocument is used in mongo but not in orm. And I could write my own persistence bundle with some others keys.

So currently any keys or values are allowed and IMHO should be allowed for now until something better is proposed like an Object